kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs

```diff
@@ -108,6 +109,7 @@ impl TypeExt for Type {
     }
 }
 
+#[allow(unused)]
```

Comment:
Is this necessary? I see that the trait is used below.

Reply:
I get the following warnings without it:
```
warning: method `is_unit_pointer` is never used
  --> kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:55:8
   |
50 | pub trait TypeExt {
   |           ------- method in this trait
...
55 |     fn is_unit_pointer(&self) -> bool;
   |        ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: trait `ExprExt` is never used
   --> kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:112:7
    |
112 | trait ExprExt {
    |       ^^^^^^^

warning: `kani-compiler` (bin "kani-compiler") generated 2 warnings
```

Reply:
I see. In that case, I would suggest deleting the unused code (e.g. `ExprExt` trait and the `impl` block).